manifests/plugin/container_gateway.pp

```diff
@@ -22,13 +36,34 @@
   Boolean $enabled = true,
   Foreman_proxy::ListenOn $listen_on = 'https',
   Stdlib::HTTPUrl $pulp_endpoint = "https://${facts['networking']['fqdn']}",
+  String $database_backend = 'postgres',
   Stdlib::Absolutepath $sqlite_db_path = '/var/lib/foreman-proxy/smart_proxy_container_gateway.db',
   Optional[Integer] $sqlite_timeout = undef,
+  Boolean $manage_postgresql = true,
+  Stdlib::Host $postgresql_host = 'localhost',
+  Stdlib::Port $postgresql_port = 5432,
+  String $postgresql_database = 'container_gateway',
+  String $postgresql_user = 'foreman-proxy',
+  String $postgresql_password = extlib::cache_data('container_gateway_cache_data', 'db_password', extlib::random_password(32))
 ) {
   foreman_proxy::plugin::module { 'container_gateway':
     version   => $version,
     enabled   => $enabled,
     feature   => 'Container_Gateway',
     listen_on => $listen_on,
   }
+
+  if $foreman_proxy::plugin::container_gateway::manage_postgresql {
+    include postgresql::server
+    postgresql::server::db { $foreman_proxy::plugin::container_gateway::postgresql_database:
+      user     => $foreman_proxy::plugin::container_gateway::postgresql_user,
+      password => postgresql::postgresql_password(
+        $foreman_proxy::plugin::container_gateway::postgresql_user,
+        $foreman_proxy::plugin::container_gateway::postgresql_password
+      ),
+      encoding => 'utf8',
+      locale   => 'en_US.utf8',
+      require  => Package['glibc-langpack-en'],
```

Comment:
It now fails on this. You should add `ensure_packages(['glibc-langpack-en'])`.

The background is that for the `en_US.utf8` locale is included in the `glibc-langpack-en` and that's not present in a (very) minimal installation.

An alternative we don't have elsewhere (yet?) is to use the `C.utf8` locale, which is guaranteed to be present on EL8+. Since we're only now starting to deploy this service, I think it's a good chance to start with that.

@evgeni any thoughts on that?

Reply:
`C` vs `en_US`? I'd def. prefer `C` if we can.